### ISSUE TITLE
For distillation: add option to use top-k logits

### ISSUE THREAD

**Issue by HannanJaved** [whittle-org/whittle#340]
**Is your feature request related to a problem? Please describe.**
We currently only support full logits; or top-k only if we have top-k logits already stored

**Describe the solution you'd like**
Add an option/arg to use only top_k logits for distillation instead of full logits
